### DISCUSSION TITLE
“Flat" Attributes vs Object's Attributes

### DISCUSSION THREAD

**Discussion by nbareil** [MISP/MISP#9113] [Q&A] [answered]
Hello all

I was wondering what was the best practice regarding attributes: For example, we have an enrichment service that "expand"  hashes (you give a MD5 and it will add SHA1 and SHA256 attributes to the event).

As of today, we push them as "flat attributes" (in `Event.Attribute`) with a comment stating it is linked to MD5 XZY. We would like to upgrade this service and create MISP `file` objects instead.

Then, what would be the best practice?

1. Delete the initial flat attributes and create only the MISP object
2. Create the MISP object **and** create the flat attributes as well
3. Keep the initial flat attribute and create the MISP object
4. I don't think this is possible, but would it be possible to have the MISP object _referencing_ a flat attribute?

Bonus question: If we do solution 1, would there be any synchronisation issues at the end of the day? (If we are modifying imported events)

On top of that, there is a wider issue with the usual "consuming tool" that do not checks objects' attributes but only flat attributes. is it something being considered in the future developments?

Thanks!

**Comment by adulau** [comment, marked as answer]
Usually the practice is `3` creating new MISP objects.  Many MISP expansion modules create a relationship between the objects and the original attributes. An [example module](https://github.com/MISP/misp-modules/blob/main/misp_modules/modules/expansion/hashlookup.py#L32).
